srml/babe/src/lib.rs

```diff
@@ -216,17 +233,26 @@ impl<T: Trait> OnTimestampSet<T::Moment> for Module<T> {
 	fn on_timestamp_set(_moment: T::Moment) { }
 }
 
+impl<T: Trait> session::ShouldEndSession<u64> for Module<T> {
+	fn should_end_session(_now: u64) -> bool {
+		LastSlotInEpoch::get()
```

Comment:
```suggestion
		LastSlotInEpoch::take()
```

I think this should be safe, as when we return `true` here, we probably see a `on_new_session` in the next block?

Reply:
@bkchr that's right